storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -1159,6 +1161,18 @@
           (check-authorization! nimbus storm-name topology-conf "deactivate"))
         (transition-name! nimbus storm-name :inactivate true))
 
+      (debug [this storm-name component-id enable?]
+        (let [storm-cluster-state (:storm-cluster-state nimbus)
+              storm-id (get-storm-id storm-cluster-state storm-name)
+              topology-conf (try-read-storm-conf conf storm-id)
+              storm-base-updates (assoc {} :component->debug (if (empty? component-id) {storm-id enable?} {component-id enable?}))]
+          (check-authorization! nimbus storm-name topology-conf "debug")
+          (when-not storm-id
+            (throw (NotAliveException. storm-name)))
+          (log-message "Nimbus setting debug to " enable? " for storm-name " storm-name " storm-id " storm-id)
```

Comment:
The log statement should also include component names for which debug is being enabled otherwise person reading the statement can falsely assume the debug is being enabled for the whole topology.